File: bbarena-model/src/main/java/net/sf/bbarena/model/team/PlayerPitchStatus.java
package net.sf.bbarena.model.team;

/**
 * Created by Teg on 26/12/2016.
 */
public enum PlayerPitchStatus {
    STANDING, PRONE, STUNNED;

    public static PlayerPitchStatus getPitchStatus(int pitchStatus) {
        PlayerPitchStatus res = null;
        if (pitchStatus >= 1 && pitchStatus <= 8) {
            res = PlayerPitchStatus.values()[pitchStatus];
        }
        return res;
    }
}

File: src/main/kotlin/bbarena/model/team/PlayerPitchStatus.kt
package bbarena.model.team

import bbarena.model.exception.PitchException

fun getPitchStatus(pitchStatus: Int) =
    if (pitchStatus in 0..2) PlayerPitchStatus.values()[pitchStatus]
    else throw PitchException("Pitch status $pitchStatus is not supported")

enum class PlayerPitchStatus {
    STANDING, PRONE, STUNNED
}
